mycroft/tts/cache_handler.py

```diff
@@ -45,28 +45,26 @@
 cache_dialog_path = [res_path, wifi_setup_path]
 
 
-def generate_cache_text(cache_audio_dir, cache_text_file):
+def generate_cache_text(cache_text_file):
```

Comment:
this is deprecating the old API, maybe throw a deprecation warning and support the argument for a while

Reply:
Ah yeah good point. 

Should we also make them private methods and only expose a single `cache_handler`? Or are there uses for these independently?

Actually it would definitely need a little more refactoring to be properly generic.

Reply:
i'm not sure this even fits in mycroft-core tbh, feels like an helper script that should be there in the first place.

but now that its there, there might be users in the wild depending on it, eg, OVOS could be using that to generate its own cache (it isn't afaik)

making them private would also break the api compatibility, so still needs a deprecation warning, that way next major version tells everyone to stop depending on it, and the one after can make them private/remove them

Reply:
Yeah, agree we shouldn't break the API. 

Given it's currently hardcoded to Mimic2, we could extract it to that plugin. At the moment this seems like the most likely course of action to me.

Alternatively, if there is interest and someone willing to do it, they could refactor it for use in any TTS in which case we could leave it in core as a common method.